manifests/server/binarylog.pp

```diff
@@ -3,19 +3,22 @@
 
   $options = $mysql::server::options
   $includedir = $mysql::server::includedir
+  $manage_binlog_dir = $mysql::server::manage_binlog_dir
 
   $logbin = pick($options['mysqld']['log-bin'], $options['mysqld']['log_bin'], false)
 
-  if $logbin {
-    $logbindir = mysql_dirname($logbin)
+  if $manage_binlog_dir == true {
+    if $logbin {
+      $logbindir = mysql_dirname($logbin)
 
-    #Stop puppet from managing directory if just a filename/prefix is specified
-    if $logbindir != '.' {
```

Comment:
This conditional already disables the binlog directory management; it would probably be simple to just extend it to check `$manage_binlog_dir` here.

Reply:
Actually the conditional above, `if $logbin {` is a better spot...